### DISCUSSION TITLE
Does `useContractRead` ever use the injected provider?

### DISCUSSION THREAD

**Discussion by gpxl-dev** [wevm/wagmi#855] [Q&A] [answered]
As per the title, I'm finding that `useContractRead` is always using a provider that I've set up in `configureChains` regardless or whether or not I've connected my wallet.

Is this expected?

Ideally I'd like to prefer to use the RPC from the connected wallet.

**Comment by tmm** [comment, marked as answer]
It always uses the client's provider. This tends to be a lot more reliable than using wallet RPCs. Often, when using `configureChains`, connectors (e.g. `WalletConnectConnector`) end up using the same RPC endpoints as your provider.

**Reply by gpxl-dev** [reply]
@tmm coming back to this - is there an easy way to force use of the user's wallet RPC?
For apps with an extremely high volume of traffic, it centralises requests to a single API key which ultimately makes things _less_ reliable if the rate limit gets hit. It would be nice to have the option to fall back to the wallet's RPC.

**Reply by tmm** [reply]
See https://github.com/wagmi-dev/wagmi/discussions/1281#discussioncomment-4162113
